### PR TITLE
[RAC-6339] iPDU firmware update workflow define.

### PR DESCRIPTION
**Pull request by HQebupt**
## iPDU:  implement the Firmware Update coding
In order to support the switch iPDU of ServerTech.  We use the 'CWG-16H2C454' model PDU to test.

The final solution is "SFTP Client put". RackHD provide a workflow to update iPDU firmware, named 'Graph.SftpOperation'.

- Relate PR : [on-tasks-536](https://github.com/RackHD/on-tasks/pull/536)

We can add a firmware update workflow by command: 

```
curl -X POST -H 'Content-Type: application/json' localhost:8080/api/2.0/nodes/{nodeID}/workflows -d '{
    "name": "Graph.SftpOperation",
    "options": {
        "defaults": {
             "isPDU": "true",
             "fileSource": "{localFile}",
             "fileDestination": "{remoteFile}"
        }
      
    }
} ' |jq '.' | grep 'instanceId'
```

@panpan0000 @anhou @pengz1

### CODE REVIEW

File: lib/graphs/iPDU-firmware-update-by-sftp-graph.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2017, EMC, Inc.
+
+"use strict";
+
+console.log('==== call sftp-demo-graph.js');
```
Comment (pengz1) [inline lib/graphs/iPDU-firmware-update-by-sftp-graph.js]: Remove this line

File: lib/graphs/iPDU-firmware-update-by-sftp-graph.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2017, EMC, Inc.
+
+"use strict";
+
+console.log('==== call sftp-demo-graph.js');
+module.exports = {
+  friendlyName: "Sftp client operation",
+  injectableName: "Graph.SftpOperation",
```
Comment (pengz1) [inline lib/graphs/iPDU-firmware-update-by-sftp-graph.js]: Please change your name to "ipdu-firmware-update" related.

File: lib/graphs/iPDU-firmware-update-by-sftp-graph.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+                options: {
+                    isPDU: "true",
+                    fileSource: null,
+                    fileDestination: null
+                },
+                properties: {}
+            }
+    }
+  ]
+};
```
Comment (pengz1) [inline lib/graphs/iPDU-firmware-update-by-sftp-graph.js]: You need a unit-test document for this graph definition.

File: package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "//": "Copyright 2017, Dell EMC, Inc.",
   "name": "on-taskgraph",
-  "version": "2.28.0",
+  "version": "2.29.0",
```
Comment (pengz1) [inline package.json]: It is not necessary to change version information.

Comment (panpan0000) [inline package.json]: please rebase again to remove this change.

File: debian/changelog [hunk 5]
```diff
@@ -36,5 +36,6 @@ on-taskgraph (2.28.0) UNRELEASED; urgency=low
   * release 2.26.0
   * release 2.27.0
   * release 2.28.0
+  * release 2.29.0
 
- -- rackhd <rackhd@emc.com>  Fri, 20 Oct 2017 21:02:19 -0400
+ -- rackhd <rackhd@emc.com>  Wed, 25 Oct 2017 22:54:43 -0400
```
Comment (pengz1) [inline debian/changelog]: It is not necessary to change this file.

File: debian/changelog [hunk 6]
```diff
@@ -1,4 +1,4 @@
-on-taskgraph (2.28.0) UNRELEASED; urgency=low
```
Comment (panpan0000) [inline debian/changelog]: please rebase again to remove this change.

File: lib/graphs/iPDU-firmware-update-by-sftp-graph.js (hunk 1 above)
Comment (panpan0000) [inline lib/graphs/iPDU-firmware-update-by-sftp-graph.js]: remove debug code

**Comment by pengz1** [comment]
If possible, you can create a schema documents for the task you changed.

File: spec/lib/graphs/iPDU-firmware-update-by-sftp-graph-spec.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+//Copyright © 2017 Dell Inc. or its subsidiaries. All Rights Reserved.
+/* jshint node:true */
+
+'use strict';
```
Comment (panpan0000) [inline spec/lib/graphs/iPDU-firmware-update-by-sftp-graph-spec.js]: +0.   @pengz1  @anhou  to review the unit-test code

File: lib/graphs/examples/iPDU-firmware-update-by-sftp-graph.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2017, EMC, Inc.
```
Comment (anhou) [inline lib/graphs/examples/iPDU-firmware-update-by-sftp-graph.js]: //Copyright © 2017 Dell Inc. or its subsidiaries. All Rights Reserved.

Comment (anhou) [inline lib/graphs/examples/iPDU-firmware-update-by-sftp-graph.js]: //Copyright © 2017 Dell Inc. or its subsidiaries. All Rights Reserved.

**Pull request merged**